### PR TITLE
build: Downgrade semantic release to support nodev18

### PR DESCRIPTION
**Pull request by DaevMithran**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,17 +47,17 @@
   },
   "devDependencies": {
     "@semantic-release/changelog": "^6.0.3",
-    "@semantic-release/commit-analyzer": "^11.1.0",
+    "@semantic-release/commit-analyzer": "^10.0.4",
```
Comment (ankurdotb) [inline package.json:50]: Don't think this downgrade is needed. Node v18 support is up to v11 on this. v12 is the one that drops Node v18 support

File: package.json [hunk 2]
```diff
@@ -47,17 +47,17 @@
...
     "@semantic-release/git": "^10.0.1",
     "@semantic-release/github": "^9.2.6",
-    "@semantic-release/npm": "^11.0.2",
-    "@semantic-release/release-notes-generator": "^12.1.0",
+    "@semantic-release/npm": "^10.0.3",
```
Comment (ankurdotb) [inline package.json:53]: Node v18 support was dropped in v12, so v11 is fine

File: package.json [hunk 3]
```diff
@@ -47,17 +47,17 @@
...
+    "@semantic-release/release-notes-generator": "^11.0.7",
```
Comment (ankurdotb) [inline package.json:54]: Node v18 support was dropped in v13, so v12 is fine.

File: package.json [hunk 4]
```diff
@@ -47,17 +47,17 @@
...
     "@types/long": "^5.0.0",
-    "@types/node": "^20.11.20",
-    "conventional-changelog-conventionalcommits": "^7.0.2",
+    "@types/node": "^20.16.2",
```
Comment (ankurdotb) [inline package.json:56]: This should be v18

File: package.json [hunk 5]
```diff
@@ -47,17 +47,17 @@
...
+    "conventional-changelog-conventionalcommits": "^8.0.0",
```
Comment (ankurdotb) [inline package.json:57]: v7 is fine, v8 drops Node v18 support

File: package.json [hunk 6]
```diff
@@ -47,17 +47,17 @@
...
     "loose-ts-check": "^2.0.0",
-    "semantic-release": "^23.0.2",
-    "typescript": "^5.3.3"
+    "semantic-release": "^21.1.1",
```
Comment (ankurdotb) [inline package.json:59]: This is also fine at 23.x

File: package.json [hunk 7]
```diff
@@ -47,17 +47,17 @@
...
+    "typescript": "^5.5.4"
```
Comment (ankurdotb) [inline package.json:60]: See if this supports node v18 or what last version does

**Review by ankurdotb — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (DaevMithran) [reply package.json:50]: v11 supports only `node >= 18.9.0` not the versions before that

File: package.json (hunk 2 above)
Comment (DaevMithran) [reply package.json:53]: Same as above

File: package.json (hunk 6 above)
Comment (DaevMithran) [reply package.json:59]: V23.x only support node >= 20

File: package.json (hunk 1 above)
Comment (ankurdotb) [reply package.json:50]: That's fine, you can set the engine version higher in package.json. Node v18 LTS is already on 18.18.x. Setting it to 18.9.x and higher is fine

File: package.json (hunk 7 above)
Comment (DaevMithran) [reply package.json:60]: It does

**Review by ankurdotb — APPROVED** [review]

**Pull request merged**